internal/engine/actions/alert/pull_request_comment/pull_request_comment.go

```diff
@@ -0,0 +1,262 @@
+		gh:         gh,
+		reviewCfg:  reviewCfg,
+		setting:    setting,
+	}, nil
+}
+
+// Class returns the action type of the PR comment alert engine
+func (alert *Alert) Class() interfaces.ActionType {
+	return alert.actionType
+}
+
+// Type returns the action subtype of the PR comment alert engine
+func (_ *Alert) Type() string {
+	return AlertType
+}
+
+// GetOnOffState returns the alert action state read from the profile
+func (alert *Alert) GetOnOffState() models.ActionOpt {
+	return models.ActionOptOrDefault(alert.setting, models.ActionOptOff)
+}
+
+// Do comments on a pull request
+func (alert *Alert) Do(
+	ctx context.Context,
+	cmd interfaces.ActionCmd,
+	entity protoreflect.ProtoMessage,
+	params interfaces.ActionsParams,
+	metadata *json.RawMessage,
+) (json.RawMessage, error) {
+	pr, ok := entity.(*pbinternal.PullRequest)
+	if !ok {
+		return nil, fmt.Errorf("expected repository, got %T", entity)
+	}
+
+	commentParams, err := getParamsForPRComment(ctx, pr, params, metadata)
+	if err != nil {
+		return nil, fmt.Errorf("error extracting parameters for PR comment: %w", err)
+	}
+
+	// Process the command based on the action setting
+	switch alert.setting {
+	case models.ActionOptOn:
+		return alert.run(ctx, commentParams, cmd)
+	case models.ActionOptDryRun:
+		return alert.runDry(ctx, commentParams, cmd)
+	case models.ActionOptOff, models.ActionOptUnknown:
+		return nil, fmt.Errorf("unexpected action setting: %w", enginerr.ErrActionFailed)
+	}
+	return nil, enginerr.ErrActionSkipped
+}
+
+func (alert *Alert) run(ctx context.Context, params *paramsPR, cmd interfaces.ActionCmd) (json.RawMessage, error) {
+	logger := zerolog.Ctx(ctx)
+
+	// Process the command
+	switch cmd {
+	// Create a review
+	case interfaces.ActionCmdOn:
+		review := &github.PullRequestReviewRequest{
+			CommitID: github.String(params.CommitSha),
+			Event:    github.String("COMMENT"),
+			Body:     github.String(alert.reviewCfg.ReviewMessage),
+		}
+
+		r, err := alert.gh.CreateReview(
+			ctx,
+			params.Owner,
+			params.Repo,
+			params.Number,
+			review,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("error creating PR review: %w, %w", err, enginerr.ErrActionFailed)
+		}
+
+		newMeta, err := json.Marshal(alertMetadata{
+			ReviewID:       r.ID,
+			SubmittedAt:    r.SubmittedAt.GetTime(),
+			PullRequestUrl: r.PullRequestURL,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("error marshalling alert metadata json: %w", err)
+		}
+
+		logger.Info().Int64("review_id", *r.ID).Msg("PR review created")
+		return newMeta, nil
+	// Dismiss the review
+	case interfaces.ActionCmdOff:
+		if params.Metadata == nil || params.Metadata.ReviewID == nil {
+			// We cannot do anything without the PR review ID, so we assume that turning the alert off is a success
+			return nil, fmt.Errorf("no PR comment ID provided: %w", enginerr.ErrActionTurnedOff)
+		}
+
+		_, err := alert.gh.DismissReview(ctx, params.Owner, params.Repo, params.Number, *params.Metadata.ReviewID,
```

Comment:
Do I read this correctly that the ReviewID is what would differentiate alerts/reviews between different ruletypes?

Reply:
Yes, because at this point we do one review/comment per ruletype. Once we combine all Minder comments into a single review, then we will only have one reviewId per entity